lib/ecto/query/builder.ex

```diff
@@ -777,7 +777,7 @@ defmodule Ecto.Query.Builder do
     do: do_literal(value, expected, quoted_type(value, vars))
 
   defp do_literal(value, _, current) when current in @always_tagged,
-    do: {:%, [], [Ecto.Query.Tagged, {:%{}, [], [value: value, type: current]}]}
+    do: {:%, [], [Ecto.Query.Tagged, {:%{}, [], [value: value, type: normalize_type(value, current)]}]}
```

Comment:
I preferred to create a new function `normalize_type` here. This way, I can match on the `:binary` current type so if, in future, `@always_tagged` were to receive new values other than `:binary` it wouldn't silently be converted into `:bitstring` but would throw an error.

I tried to put a catch all definition of `normalize_type`, but the compiler warns me that it can never match because `normalize_type(value, :binary)` clauses completely cover the type.

Reply:
i think it's ok. if we add new types then we would see the error when it fails to normalize.

sorry just one thing. could you please do `quote do: ...` instead of the raw AST? it just makes it easier to read/remember what is happening when we have to come back to it later

Reply:
Changed to `quote do: ...` but I had to also change the test to assert a pattern matching instead of an equality. 

One of the last test runs failed and I don't know why. At first glance, it seems unrelated to my changes..